<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to easyEWAS • easyEWAS</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to easyEWAS">
<meta property="og:description" content="easyEWAS">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">easyEWAS</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/easyEWAS.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to easyEWAS</h1>
            
            <h4 data-toc-skip class="date">2024-07-01</h4>
      
      
      <div class="hidden name"><code>easyEWAS.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="summary">
<strong>Summary</strong><a class="anchor" aria-label="anchor" href="#summary"></a>
</h2>
<p><strong>easyEWAS</strong> is a user-friendly R package designed for
conducting epigenome-wide association studies (EWAS). It provides
multiple statistical analysis methods tailored for common EWAS study
designs and accelerates computation efficiency through parallel
computing. Specifically, easyEWAS emphasizes downstream analyses of
EWAS, including internal validation using bootstrap, functional
enrichment analysis of differentially methylated genes, and result
visualization. easyEWAS is compatible with currently most cited Illumina
high-density microarray platforms for genome-wide methylation profiling,
including 27K, 450K, EPIC V1, EPIC V2, and MSA arrays.</p>
<p><img src="hex-easyEWAS.png" width="25%" style="display: block; margin: auto;"></p>
</div>
<div class="section level2">
<h2 id="workflow">
<strong>Workflow</strong><a class="anchor" aria-label="anchor" href="#workflow"></a>
</h2>
<div class="section level3">
<h3 id="installation">
<strong>Installation</strong><a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>Users can install the easyEWAS package by running the following
code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"ytwangZero/easyEWAS"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="prepare-data-files">
<strong>Prepare Data Files</strong><a class="anchor" aria-label="anchor" href="#prepare-data-files"></a>
</h3>
<p><strong>Sample data:</strong> Users need to prepare a sample file to
store sample information, such as sample ID, exposure variables,
covariates, etc. easyEWAS only supports .csv and .xlsx files, the sample
format is as follows:
<img src="sampledata.png" width="50%" style="display: block; margin: auto;"></p>
<p><strong>Methylation data:</strong> Users also need to prepare a
methylation file, requiring each row to represent a CpG site and each
column to represent a sample. easyEWAS only supports .csv and .xlsx
files, the data format is as follows:
<img src="methydata.png" width="70%" style="display: block; margin: auto;"></p>
</div>
<div class="section level3">
<h3 id="example">
<strong>Example</strong><a class="anchor" aria-label="anchor" href="#example"></a>
</h3>
<div class="section level4">
<h4 id="Library-the-package">
<strong>Library the package</strong><a class="anchor" aria-label="anchor" href="#library-the-package"></a>
</h4>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">easyEWAS</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="step-1-initialize-the-ewas-module">
<strong>Step 1 Initialize the EWAS module</strong><a class="anchor" aria-label="anchor" href="#step-1-initialize-the-ewas-module"></a>
</h4>
<p>Here, take the easyEWAS internal dataset as an example. After loading
easyEWAS, the user needs to use the <code><a href="../reference/initEWAS.html">initEWAS()</a></code> function to
initialize an analysis module. This will generate a folder named
<em>EWASresult</em> under the working path specified by the user to
store the EWAS analysis results. The function also returns an R6 object
integrating all information, so the user needs to assign the result of
the function to a new name, such as <code>res</code>. If default, the
results will be stored in the current working path.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/initEWAS.html">initEWAS</a></span><span class="op">(</span>outpath <span class="op">=</span> <span class="st">"default"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Complete initializing the EWAS module! Your EWAS results will be stored in D:/R/Rwork/Rpackage/easyEWAS/vignettes/EWASresult</span></span>
<span><span class="co">#&gt; 2024-06-29 15:23:06.17881</span></span>
<span><span class="co">#&gt; 0.11 sec elapsed</span></span></code></pre></div>
<p> </p>
</div>
<div class="section level4">
<h4 id="step-2-load-the-data-files">
<strong>Step 2 Load the Data Files</strong><a class="anchor" aria-label="anchor" href="#step-2-load-the-data-files"></a>
</h4>
<p>Users can upload sample data and methylation data in two ways. One is
to specify the path where the file is located, and the other is to
assign the data set in the R environment directly to the parameters
<em>ExpoData</em> and <em>MethyData</em>. If default, the data set
inside the R package is called.</p>
<p>Users can view the sample data and the methylation data for the
examples by running the following codes:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"sampledata"</span>, package <span class="op">=</span> <span class="st">"easyEWAS"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sampledata</span><span class="op">)</span></span>
<span><span class="co">#&gt;   SampleName var cov1     cov2</span></span>
<span><span class="co">#&gt; 1 ZE10000001  76    2 20.96810</span></span>
<span><span class="co">#&gt; 2 ZE10000002  66    2 26.78373</span></span>
<span><span class="co">#&gt; 3 ZE10000003  71    2 28.26081</span></span>
<span><span class="co">#&gt; 4 ZE10000004  81    1 22.54459</span></span>
<span><span class="co">#&gt; 5 ZE10000005  70    1 21.66550</span></span>
<span><span class="co">#&gt; 6 ZE10000006  68    2 16.94033</span></span></code></pre></div>
<p>Here we view the first 6 columns of the methylation data inside the R
package.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"methydata"</span>, package <span class="op">=</span> <span class="st">"easyEWAS"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">methydata</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 6</span></span></span>
<span><span class="co">#&gt;   probe           ZE10000001 ZE10000002 ZE10000003 ZE10000004 ZE10000005</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> cg01444397_BC11     0.047<span style="text-decoration: underline;">7</span>     0.027<span style="text-decoration: underline;">8</span>     0.028<span style="text-decoration: underline;">5</span>     0.067<span style="text-decoration: underline;">1</span>     0.044<span style="text-decoration: underline;">8</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> cg12110931_TC11     0.439      0.432      0.513      0.451      0.600 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> cg12258811_TC21     0.420      0.477      0.528      0.536      0.643 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> cg08764927_TC21     0.821      0.823      0.887      0.872      0.913 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> cg25514503_TC21     0.950      0.951      0.958      0.962      0.960 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> cg24454741_TC21     0.881      0.896      0.907      0.885      0.865</span></span></code></pre></div>
<p>Note the <code>input</code> parameter, which needs to be assigned the
<em>res</em> generated in the previous step. If the user chooses to read
data from external sources, the file paths need to be specified in the
parameters <code>ExpoPath</code> and <code>MethyPath</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/loadEWAS.html">loadEWAS</a></span><span class="op">(</span>input <span class="op">=</span> <span class="va">res</span>,</span>
<span>                ExpoData <span class="op">=</span> <span class="st">"default"</span>,</span>
<span>                MethyData <span class="op">=</span> <span class="st">"default"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Sample data and methylation data for example have been loaded! </span></span>
<span><span class="co">#&gt; 2024-06-29 15:23:06.34943</span></span>
<span><span class="co">#&gt; 0.03 sec elapsed</span></span></code></pre></div>
<p> </p>
</div>
<div class="section level4">
<h4 id="step-3-transform-the-variable-types">
<strong>Step 3 Transform the variable types</strong><a class="anchor" aria-label="anchor" href="#step-3-transform-the-variable-types"></a>
</h4>
<p>This is an optional step where the user can transform certain
variables in the sample dataset to a specified type. The function
support both factor and numeric types.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/transEWAS.html">transEWAS</a></span><span class="op">(</span>input <span class="op">=</span> <span class="va">res</span>, Vars <span class="op">=</span> <span class="st">"cov1"</span>,TypeTo <span class="op">=</span> <span class="st">"factor"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Variable types have been transformed successfully!</span></span>
<span><span class="co">#&gt; 0.02 sec elapsed</span></span></code></pre></div>
<p> </p>
</div>
<div class="section level4">
<h4 id="step-4-peform-the-ewas-analysis">
<strong>Step 4 Peform the EWAS analysis</strong><a class="anchor" aria-label="anchor" href="#step-4-peform-the-ewas-analysis"></a>
</h4>
<p>This is the core step of the analysis. The user can select models
based on the analysis requirements, including linear regression models,
linear mixed effects models, and Cox proportional risk models. The
function also supports parallel operations, and the number of physical
cores required for operations can be specified in the <code>core</code>
parameter. The result will be automatically stored in a custom named.csv
file. For the meaning of each parameter, refer to the manual of the R
package.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/startEWAS.html">startEWAS</a></span><span class="op">(</span>input <span class="op">=</span> <span class="va">res</span>,</span>
<span>                filename <span class="op">=</span> <span class="st">"default"</span>,</span>
<span>                chipType <span class="op">=</span> <span class="st">"EPICV2"</span>, <span class="co"># "EPICV2", "EPICV1", "450K", "27K", "MSA"</span></span>
<span>                model <span class="op">=</span> <span class="st">"lm"</span>, <span class="co"># "lm", "lmer", "cox"</span></span>
<span>                expo <span class="op">=</span> <span class="st">"default"</span>,</span>
<span>                cov <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                random <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                time <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                status <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                adjustP <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                core <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="co">#&gt; It will take some time, please be patient...</span></span>
<span><span class="co">#&gt; Start the EWAS analysis...</span></span>
<span><span class="co">#&gt; Start multiple testing correction ...</span></span>
<span><span class="co">#&gt; Start CpG sites annotation ...</span></span>
<span><span class="co">#&gt; EWAS analysis has been completed! </span></span>
<span><span class="co">#&gt; You can find results in D:/R/Rwork/Rpackage/easyEWAS/vignettes/EWASresult.</span></span>
<span><span class="co">#&gt; 2024-06-29 15:23:17.078925</span></span>
<span><span class="co">#&gt; 10.64 sec elapsed</span></span></code></pre></div>
<p>Users can view the analysis results by running the following code.
For linear regression and linear mixed-effects models, when the exposure
variable is continuous, the coefficient value (per unit, per IQR, per
SD), standard error, significance P value, adjusted P value, and
annotation information for the corresponding site are provided. For
categorical variables, the coefficient for each class compared to the
reference group is provided. For Cox proportional hazards models, HR
values and confidence intervals are also provided.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">result</span><span class="op">)</span></span>
<span><span class="co">#&gt;             probe          BETA    BETA_perSD   BETA_perIQR           SE</span></span>
<span><span class="co">#&gt; 1 cg01444397_BC11  5.447755e-04  0.0050873534  0.0083078263 1.582456e-04</span></span>
<span><span class="co">#&gt; 2 cg12110931_TC11  1.893556e-03  0.0176828550  0.0288767218 6.035905e-04</span></span>
<span><span class="co">#&gt; 3 cg12258811_TC21  1.606180e-04  0.0014999213  0.0024494240 5.281461e-04</span></span>
<span><span class="co">#&gt; 4 cg08764927_TC21 -1.160395e-04 -0.0010836278 -0.0017696022 4.660786e-04</span></span>
<span><span class="co">#&gt; 5 cg25514503_TC21  2.532167e-05  0.0002364649  0.0003861554 8.094542e-05</span></span>
<span><span class="co">#&gt; 6 cg24454741_TC21 -9.076997e-05 -0.0008476499 -0.0013842420 1.254157e-04</span></span>
<span><span class="co">#&gt;       SE_perSD   SE_perIQR         PVAL       FDR Bonfferoni chr     pos</span></span>
<span><span class="co">#&gt; 1 0.0014777669 0.002413245 0.0008487631 0.1161862          1   1 7784390</span></span>
<span><span class="co">#&gt; 2 0.0056365937 0.009204755 0.0022531189 0.1632849          1   1 7784577</span></span>
<span><span class="co">#&gt; 3 0.0049320606 0.008054228 0.7616838900 0.9158923          1   1 7784675</span></span>
<span><span class="co">#&gt; 4 0.0043524474 0.007107699 0.8039052062 0.9290912          1   1 7784835</span></span>
<span><span class="co">#&gt; 5 0.0007559039 0.001234418 0.7550791242 0.9126204          1   1 7785010</span></span>
<span><span class="co">#&gt; 6 0.0011711875 0.001912590 0.4709417633 0.7891571          1   1 7785177</span></span>
<span><span class="co">#&gt;   relation_to_island</span></span>
<span><span class="co">#&gt; 1             Island</span></span>
<span><span class="co">#&gt; 2             Island</span></span>
<span><span class="co">#&gt; 3             Island</span></span>
<span><span class="co">#&gt; 4             Island</span></span>
<span><span class="co">#&gt; 5             Island</span></span>
<span><span class="co">#&gt; 6              Shore</span></span>
<span><span class="co">#&gt;                                                                    gene</span></span>
<span><span class="co">#&gt; 1                          PER3;PER3;PER3;PER3;PER3;PER3;PER3;PER3;PER3</span></span>
<span><span class="co">#&gt; 2                                                                      </span></span>
<span><span class="co">#&gt; 3 PER3;PER3;PER3;PER3;PER3;PER3;PER3;PER3;PER3;PER3;PER3;PER3;PER3;PER3</span></span>
<span><span class="co">#&gt; 4 PER3;PER3;PER3;PER3;PER3;PER3;PER3;PER3;PER3;PER3;PER3;PER3;PER3;PER3</span></span>
<span><span class="co">#&gt; 5                                                                      </span></span>
<span><span class="co">#&gt; 6</span></span></code></pre></div>
<p> </p>
</div>
<div class="section level4">
<h4 id="step-5-visualize-the-ewas-results">
<strong>Step 5 Visualize the EWAS results</strong><a class="anchor" aria-label="anchor" href="#step-5-visualize-the-ewas-results"></a>
</h4>
<p>Visualize EWAS results based on the <strong>CMplot</strong> package,
including Manhattan plots, QQ plots, etc. Optional parameters are the
same as those in the <code>CMplot</code> function. When a user’s
exposure variable is a categorical variable, the user can specify which
class of p-values to use for results visualization. The
<code>threshold</code> parameter is used to plot a threshold line on the
Manhattan plot.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plotEWAS.html">plotEWAS</a></span><span class="op">(</span>input <span class="op">=</span> <span class="va">res</span>,</span>
<span>                file <span class="op">=</span> <span class="st">"jpg"</span>,</span>
<span>                p <span class="op">=</span> <span class="st">"PVAL"</span>,</span>
<span>                threshold <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="co">#&gt; Ewas Visualization has completed! </span></span>
<span><span class="co">#&gt; You can find results in D:/R/Rwork/Rpackage/easyEWAS/vignettes/EWASresult.</span></span>
<span><span class="co">#&gt; 2024-06-29 15:23:17.727966</span></span>
<span><span class="co">#&gt; 0.58 sec elapsed</span></span></code></pre></div>
<p>Here is the Quantile-Quantile plot of the EWAS results.
<img src="qq.png" width="50%" style="display: block; margin: auto;"></p>
<p> </p>
</div>
<div class="section level4">
<h4 id="step-6-bootstrap-for-internal-validation">
<strong>Step 6 Bootstrap for internal validation</strong><a class="anchor" aria-label="anchor" href="#step-6-bootstrap-for-internal-validation"></a>
</h4>
<p>Bootstrap is a statistical resampling technique used for internal
validation and estimation of uncertainty in statistical models. It’s
particularly useful when the data is limited or when the underlying
distribution of the data is not well known. Bootstrap can be a valuable
tool for validating results from the EWAS. Users can run the following
code to obtain the results of internal validation.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bootEWAS.html">bootEWAS</a></span><span class="op">(</span>input <span class="op">=</span> <span class="va">res</span>,</span>
<span>                filterP <span class="op">=</span> <span class="st">"PVAL"</span>, <span class="co"># FDR, Bonferroni</span></span>
<span>                cutoff <span class="op">=</span> <span class="fl">0.001</span>,</span>
<span>                bootCI <span class="op">=</span> <span class="st">"perc"</span>,</span>
<span>                times <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="co">#&gt; Bootstrap analysis for cg01444397_BC11...</span></span>
<span><span class="co">#&gt; Bootstrap analysis for cg01937505_BC21...</span></span>
<span><span class="co">#&gt; Bootstrap analysis for cg14151354_BC21...</span></span>
<span><span class="co">#&gt; Bootstrap analysis for cg13515269_TC21...</span></span>
<span><span class="co">#&gt; Bootstrap analysis for cg16047471_BC21...</span></span>
<span><span class="co">#&gt; Bootstrap analysis for cg06617514_TC21...</span></span>
<span><span class="co">#&gt; Bootstrap analysis for cg00200653_TC11...</span></span>
<span><span class="co">#&gt; Bootstrap analysis for cg24867447_BC21...</span></span>
<span><span class="co">#&gt; Bootstrap analysis for cg25069388_BC21...</span></span>
<span><span class="co">#&gt; Bootstrap for internal validation has been completed !</span></span>
<span><span class="co">#&gt; You can find results in D:/R/Rwork/Rpackage/easyEWAS/vignettes/EWASresult.</span></span>
<span><span class="co">#&gt; 2024-06-29 15:23:18.401566</span></span>
<span><span class="co">#&gt; 0.61 sec elapsed</span></span></code></pre></div>
<p>Users can select CpG sites to validate in two ways. One is to set the
P-value column (PVAL, FDR, Bonfferoni) to be filtered and its filtering
cutoff value (default is 0.05), and the other is to specify the name of
the site to be validated directly in the <code>CpGs</code> parameter.
The result is stored in <code>res$bootres</code>. The function
ultimately provides the original statistical values for each CpG site
alongside their bootstrap-derived confidence intervals.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">bootres</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 4</span></span></span>
<span><span class="co">#&gt;   probe            original  lower_CI  upper_CI</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> cg01444397_BC11  0.000<span style="text-decoration: underline;">545</span>  0.000<span style="text-decoration: underline;">118</span>  0.000<span style="text-decoration: underline;">958</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> cg01937505_BC21 -<span style="color: #BB0000;">0.001</span><span style="color: #BB0000; text-decoration: underline;">11</span>  -<span style="color: #BB0000;">0.001</span><span style="color: #BB0000; text-decoration: underline;">77</span>  -<span style="color: #BB0000;">0.000</span><span style="color: #BB0000; text-decoration: underline;">522</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> cg14151354_BC21 -<span style="color: #BB0000;">0.000</span><span style="color: #BB0000; text-decoration: underline;">749</span> -<span style="color: #BB0000;">0.001</span><span style="color: #BB0000; text-decoration: underline;">13</span>  -<span style="color: #BB0000;">0.000</span><span style="color: #BB0000; text-decoration: underline;">307</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> cg13515269_TC21 -<span style="color: #BB0000;">0.001</span><span style="color: #BB0000; text-decoration: underline;">42</span>  -<span style="color: #BB0000;">0.002</span><span style="color: #BB0000; text-decoration: underline;">27</span>  -<span style="color: #BB0000;">0.000</span><span style="color: #BB0000; text-decoration: underline;">712</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> cg16047471_BC21 -<span style="color: #BB0000;">0.001</span><span style="color: #BB0000; text-decoration: underline;">90</span>  -<span style="color: #BB0000;">0.002</span><span style="color: #BB0000; text-decoration: underline;">68</span>  -<span style="color: #BB0000;">0.000</span><span style="color: #BB0000; text-decoration: underline;">972</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> cg06617514_TC21 -<span style="color: #BB0000;">0.001</span><span style="color: #BB0000; text-decoration: underline;">45</span>  -<span style="color: #BB0000;">0.002</span><span style="color: #BB0000; text-decoration: underline;">29</span>  -<span style="color: #BB0000;">0.000</span><span style="color: #BB0000; text-decoration: underline;">869</span></span></span></code></pre></div>
<p> </p>
</div>
<div class="section level4">
<h4 id="step-7-enrichment-analysis">
<strong>Step 7 Enrichment analysis</strong><a class="anchor" aria-label="anchor" href="#step-7-enrichment-analysis"></a>
</h4>
<p>easyEWAS provides two enrichment analysis methods (GO and KEGG) based
on the <strong>clusterProfiler</strong> package. Users can filter the
sites and their corresponding genes for enrichment analysis using the
<code>filterP</code> and <code>cutoff</code> parameters. The analysis
results are stored in <code>res$enrichresult</code>. Additionally, the
function can visualize the enrichment analysis results, including both
dotplot and barplot.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/enrichEWAS.html">enrichEWAS</a></span><span class="op">(</span>input <span class="op">=</span> <span class="va">res</span>,</span>
<span>                  method <span class="op">=</span> <span class="st">"GO"</span>,</span>
<span>                  filterP <span class="op">=</span> <span class="st">"PVAL"</span>,</span>
<span>                  cutoff <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>                  plot <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                  plotType <span class="op">=</span> <span class="st">"dot"</span>,</span>
<span>                  plotcolor <span class="op">=</span> <span class="st">"pvalue"</span>,</span>
<span>                  showCategory <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="co">#&gt; It will take some time, please be patient...</span></span>
<span><span class="co">#&gt; Start enrichment analysis ...</span></span>
<span><span class="co">#&gt; Start result visualization ...</span></span>
<span><span class="co">#&gt; Saving 7.29 x 4.51 in image</span></span>
<span><span class="co">#&gt; Enrichment analysis has been completed !</span></span>
<span><span class="co">#&gt; You can find results in D:/R/Rwork/Rpackage/easyEWAS/vignettes/EWASresult.</span></span>
<span><span class="co">#&gt; 2024-06-29 15:23:39.434813</span></span>
<span><span class="co">#&gt; 20.95 sec elapsed</span></span></code></pre></div>
<p>The result is stored in <code>res$enrichres</code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">enrichres</span><span class="op">)</span></span>
<span><span class="co">#&gt;                    ID</span></span>
<span><span class="co">#&gt; GO:0048511 GO:0048511</span></span>
<span><span class="co">#&gt; GO:0032922 GO:0032922</span></span>
<span><span class="co">#&gt; GO:0018107 GO:0018107</span></span>
<span><span class="co">#&gt; GO:0018210 GO:0018210</span></span>
<span><span class="co">#&gt; GO:2000058 GO:2000058</span></span>
<span><span class="co">#&gt; GO:0007623 GO:0007623</span></span>
<span><span class="co">#&gt;                                                            Description</span></span>
<span><span class="co">#&gt; GO:0048511                                            rhythmic process</span></span>
<span><span class="co">#&gt; GO:0032922                     circadian regulation of gene expression</span></span>
<span><span class="co">#&gt; GO:0018107                          peptidyl-threonine phosphorylation</span></span>
<span><span class="co">#&gt; GO:0018210                             peptidyl-threonine modification</span></span>
<span><span class="co">#&gt; GO:2000058 regulation of ubiquitin-dependent protein catabolic process</span></span>
<span><span class="co">#&gt; GO:0007623                                            circadian rhythm</span></span>
<span><span class="co">#&gt;            GeneRatio   BgRatio       pvalue    p.adjust      qvalue</span></span>
<span><span class="co">#&gt; GO:0048511       3/5 297/18888 3.759965e-05 0.009174315 0.003285022</span></span>
<span><span class="co">#&gt; GO:0032922       2/5  70/18888 1.344212e-04 0.016399389 0.005872085</span></span>
<span><span class="co">#&gt; GO:0018107       2/5  96/18888 2.531146e-04 0.018822885 0.006739860</span></span>
<span><span class="co">#&gt; GO:0018210       2/5 106/18888 3.085719e-04 0.018822885 0.006739860</span></span>
<span><span class="co">#&gt; GO:2000058       2/5 166/18888 7.545427e-04 0.036821685 0.013184641</span></span>
<span><span class="co">#&gt; GO:0007623       2/5 205/18888 1.147290e-03 0.037938586 0.013584567</span></span>
<span><span class="co">#&gt;                          geneID Count</span></span>
<span><span class="co">#&gt; GO:0048511 PER3/BHLHE41/CSNK2A1     3</span></span>
<span><span class="co">#&gt; GO:0032922         PER3/BHLHE41     2</span></span>
<span><span class="co">#&gt; GO:0018107       PRKAG2/CSNK2A1     2</span></span>
<span><span class="co">#&gt; GO:0018210       PRKAG2/CSNK2A1     2</span></span>
<span><span class="co">#&gt; GO:2000058      CSNK1A1/CSNK2A1     2</span></span>
<span><span class="co">#&gt; GO:0007623         PER3/BHLHE41     2</span></span></code></pre></div>
<p>Here, we can obtain a bubble plot of GO enrichment analysis.
<img src="enrich.png" width="50%" style="display: block; margin: auto;"></p>
<p> </p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="reference">
<strong>Reference</strong><a class="anchor" aria-label="anchor" href="#reference"></a>
</h2>
<p><strong>1.</strong> Hall P (1992). The Bootstrap and Edgeworth
Expansion. Springer, New York. ISBN 9781461243847.<br><strong>2.</strong> <a href="https://CRAN.R-project.org/package=CMplot" class="external-link uri">https://CRAN.R-project.org/package=CMplot</a><br><strong>3.</strong> Yu, G., Wang, L. G., Han, Y., &amp; He, Q. Y.
(2012). clusterProfiler: an R package for comparing biological themes
among gene clusters. Omics : a journal of integrative biology, 16(5),
284–287. <a href="https://doi.org/10.1089/omi.2011.0118" class="external-link uri">https://doi.org/10.1089/omi.2011.0118</a></p>
<p> </p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Yuting Wang, Xu Gao.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
